lib/rrule/rule.rb

```diff
@@ -111,7 +111,9 @@ def parse_options(rule)
           raise InvalidRRule, 'COUNT must be a non-negative integer' if i < 0
           options[:count] = i
         when 'UNTIL'
-          options[:until] = Time.parse(value)
+          # The value of the UNTIL rule part MUST have the same
+          # value type as the "DTSTART" property.
+          options[:until] = @dtstart.is_a?(Date) ? Date.parse(value) : Time.parse(value)
```

Comment:
It is a little awkward that `Date.parse` won't throw an error if `value` is a time-like string...

```ruby
Date.parse('2020-01-01T12:34:01Z') # => #<Date: 2020-01-01 ((2458850j,0s,0n),+0s,2299161j)>
```

Though, I'm not sure this matters too much.